R/tt_compatibility.R

```diff
@@ -374,6 +374,30 @@ rbindl_rtables <- function(x, gap = 0, check_headers = TRUE) {
 
     lapply(x, function(xi) chk_compat_cinfos(x[[1]], xi)) ##col_info(xi)))
 
+    rbind_annot <- list(title = "", subtitles = character(), footer = character(), pf = character())
+    # Titles/footer info are (independently) retained from first object if identical or missing in all other objects
+    all_titles <- sapply(x, main_title)
+    if (all_titles[1] != "" && (all(all_titles[-1] == "") || length(unique(all_titles)) == 1)) {
+      rbind_annot[["title"]] <- all_titles[[1]]
+    }
+
+    all_sts <- lapply(x, subtitles)
+    if (length(all_sts[[1]]) > 0 && (all(sapply(all_sts[[-1]], length) == 0) ||
+      !isFALSE(Reduce(function(x, y) if (identical(x, y)) x else FALSE, all_sts)))) {
+      rbind_annot[["subtitles"]] <- all_sts[[1]]
+    }
+
+    all_footers <- lapply(x, main_footer)
+    if (length(all_footers[[1]]) > 0 && (all(sapply(all_footers[[-1]], length) == 0) ||
+      !isFALSE(Reduce(function(x, y) if (identical(x, y)) x else FALSE, all_footers)))) {
+      rbind_annot[["footer"]] <- all_footers[[1]]
+    }
+
```

Comment:
is there a way to improve readability here? also for other booleans

Reply:
Yes, if we read this carefully its doing two things, we should factor each into a function, even if its only used once, for maintainability. I'll leave what those functions would be as practice of what we talked about today. @Melkiades @ayogasekaram @edelarua

Reply:
I think now it is very clear! Thanks Emily